### DISCUSSION TITLE
Getting duplicate authors (if already existing) when adding new books

### DISCUSSION THREAD

**Discussion by AllJackedUp** [advplyr/audiobookshelf#2668] [Q&A]
Audiobookshelf v 2.72 running on Windows 11 via Docker and Portainer.

Have had no issues until this happened. I thought I would prefer having the authers listed lastname, firstname so I changed them all to this format in their bios. It worked fine, but I ended deciding to go back to firstname lastname. My author folders have always been and continue to be firstname lastname.

Now, when I add a book that has an author already in the database, it creates a new author (same name). I have to edit the book, deleted the author name and re-enter it. When I'm entering the author's name you can see two entries as you type the name in. I always choose the first one in the list and save it. That puts the book in the original author's books page as expected. I then have to delete the second author page. 

This will occur every time, regardless of the author (unless it's a new author). It's a minor inconvenience, but I'd like to get this resolved.

I appreciate your support.

**Comment by advplyr** [comment]
Can you double check that the author name is the exact same and doesn't have an extra space or punctuation somewhere?

**Reply by AllJackedUp** [reply]
Just added a new book. The author is in the database twice.  I notice that in the first instance the name is separated by a double space. The newly added instance is only separated by one space. Within the client, before or after matching, I don't see the difference.

**Reply by AllJackedUp** [reply]
![Snag_1e8274fd](https://github.com/advplyr/audiobookshelf/assets/161378137/519a3efc-3628-408d-9bf1-69f2bc123998)
![Snag_1e827fda](https://github.com/advplyr/audiobookshelf/assets/161378137/c7ecc50f-f8de-46ae-bc43-464751184206)
![Snag_1e828b73](https://github.com/advplyr/audiobookshelf/assets/161378137/708851a6-d355-4588-b294-f4a7ddb8ab1e)
![Snag_1e8295d3](https://github.com/advplyr/audiobookshelf/assets/161378137/f9cc49e1-44d8-4eb9-ac95-95ee4a160abf)

**Comment by AllJackedUp** [comment]
Editing the author's names in the database (removing the extra spaces) seems to have fixed this. 

Thanks!

**Comment by pwinnski** [comment]
FWIW, HTML renderers collapse multiple spaces to a single space when displaying text. That's why you weren't seeing the difference!
